src/docfx/build/page/Markup.cs

```diff
@@ -53,7 +56,25 @@ internal static class Markup
 
         public static MarkupResult Result => t_result;
 
-        public static MarkdownPipeline GetPipeline(MarkdownPipelineType pipelineType) => s_pipelineMapping[pipelineType];
+        public static (MarkdownDocument ast, MarkupResult result) Parse(string content)
+        {
+            if (t_result != null)
+            {
+                throw new NotImplementedException("Nested call to Markup.ToHtml");
+            }
+
+            try
+            {
+                t_result = new MarkupResult();
+                var ast = Markdown.Parse(content, s_pipelineMapping[MarkdownPipelineType.TocMarkdown]);
```

Comment:
make pipeline type and input option defaulted to Markdown?

Reply:
one flage is not enough to control the behavior.

---
In reply to: [212532763](https://github.com/dotnet/docfx/pull/3269#discussion_r212532763) [](ancestors = 212532763)